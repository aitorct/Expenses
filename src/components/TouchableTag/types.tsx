import {FlexStyle, StyleProp, TextStyle} from 'react-native';

export interface Props {
  text: string;
  onPress: () => void;
  active: boolean;
  textStyle: StyleProp<TextStyle>;
  buttonStyle: StyleProp<FlexStyle>;
}
