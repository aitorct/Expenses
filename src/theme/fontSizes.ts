const fontSizes = {
  XS: 12,
  S: 14,
  M: 16,
  L: 18,
  XL: 20,
  XXL: 24,
};

export default {
  smallText: fontSizes.S,
  text: fontSizes.M,
  largeText: fontSizes.L,
  subtitle: fontSizes.XL,
  title: fontSizes.XXL,
  actionText: fontSizes.XS,
};
