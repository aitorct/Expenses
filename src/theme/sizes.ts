const sizes = {
  s1: 1,
  s4: 4,
  s8: 8,
  s10: 10,
  s12: 12,
  s16: 16,
  s24: 24,
  s32: 32,
  s36: 36,
  s48: 48,
  s100: 100,
};

export default {
  ...sizes,
  smallIcon: sizes.s16,
  icon: sizes.s24,
  border: sizes.s1,
};
