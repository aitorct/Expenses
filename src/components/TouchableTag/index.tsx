import React from 'react';
import {StyleSheet, TouchableOpacity} from 'react-native';
import Tag from '../Tag';
import colors from '@theme/colors';
import sizes from '@theme/sizes';
import {Props} from './types';

const TouchableTag = ({
  text,
  onPress,
  active,
  textStyle,
  buttonStyle,
}: Props) => {
  const customStyles = active ? styles.active : styles.inactive;

  return (
    <TouchableOpacity style={buttonStyle} onPress={onPress}>
      <Tag
        text={text}
        textStyle={textStyle}
        style={[styles.tag, customStyles]}
      />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  tag: {
    paddingHorizontal: sizes.s16,
    paddingVertical: sizes.s12,
    borderWidth: sizes.border,
  },
  active: {
    borderColor: colors.selectedTouchableTagBorder,
  },
  inactive: {
    backgroundColor: colors.touchableTagBackground,
    borderColor: colors.touchableTagBackground,
  },
});

export default TouchableTag;
