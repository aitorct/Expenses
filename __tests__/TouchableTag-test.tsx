import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TouchableTag from '@components/TouchableTag';
import fontSizes from '@theme/fontSizes';
import sizes from '@theme/sizes';

const mockedFunction = jest.fn(() => {});

const textStyle = {fontSize: fontSizes.title};
const buttonStyle = {padding: sizes.s8};

it('Active TouchableTag renders correctly', () => {
  const tree = renderer
    .create(
      <TouchableTag
        text={'Tag test'}
        onPress={mockedFunction}
        active={true}
        textStyle={textStyle}
        buttonStyle={buttonStyle}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Inactive TouchableTag renders correctly', () => {
  const tree = renderer
    .create(
      <TouchableTag
        text={'Tag test'}
        onPress={mockedFunction}
        active={false}
        textStyle={textStyle}
        buttonStyle={buttonStyle}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
