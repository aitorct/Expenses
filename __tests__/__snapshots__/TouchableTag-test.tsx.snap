// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active TouchableTag renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7EFFA",
          "borderRadius": 36,
          "flexDirection": "row",
          "padding": 4,
          "paddingHorizontal": 8,
        },
        Array [
          Object {
            "borderWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          Object {
            "borderColor": "#74a2e2",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 24,
            },
          ]
        }
      >
        Tag test
      </Text>
    </View>
  </View>
</View>
`;

exports[`Inactive TouchableTag renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E7EFFA",
          "borderRadius": 36,
          "flexDirection": "row",
          "padding": 4,
          "paddingHorizontal": 8,
        },
        Array [
          Object {
            "borderWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          Object {
            "backgroundColor": "#7D82851A",
            "borderColor": "#7D82851A",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 24,
            },
          ]
        }
      >
        Tag test
      </Text>
    </View>
  </View>
</View>
`;
